### PR TITLE
Move quantity from item to item_packs

### PR DESCRIPTION
**Pull request by mikib0**

### CODE REVIEW

File: server/src/db/schema.ts [hunk 1]
```diff
@@ -322,6 +322,7 @@ export const itemPacks = sqliteTable(
   {
     itemId: text('item_id').references(() => item.id, { onDelete: 'cascade' }),
     packId: text('pack_id').references(() => pack.id, { onDelete: 'cascade' }),
+    quantity: integer('quantity').notNull().default(1),
```
Comment (Tadjaur) [inline server/src/db/schema.ts:325]: Please remove the default value. It is preferable to make it nullable then fill existing item_packs with quantity retrieved from item with id itemId than assuming the default quantity is one.
@taronaleksanian What do you think about this?

**Review by Tadjaur — COMMENTED** [review]
Global Comment:

For this kind of task, I recommend creating PR chains that represent the stages of the migration.
1. First Pr: Add a nullable quantity field in the item_packs (server)

2. 2nd Pr: Import existing quantity value from item to item_packs (server) 
 :warning: :warning: :warning:  You must always think about how to process existing data before deleting any row in a database. And if the data loss is intentional, you should always add an acknowledgement comment.

3. 3th Pr: 
    - remove the quantity from item and make the quantity on item_packs notNull (server)
    - apply changes on the app
    
Having small chunk of PRs will just make the review simpler and much easier.
@taronaleksanian Please add a note if I'm wrong somewhere.

File: server/src/db/schema.ts [hunk 2]
```diff
@@ -192,6 +192,7 @@ export const itemPackTemplates = sqliteTable(
     packTemplateId: text('pack_template_id').references(() => packTemplate.id, {
       onDelete: 'cascade',
     }),
+    quantity: integer('quantity').notNull().default(1),
```
[the remark this answers is not in the document]
Comment (taronaleksanian) [reply server/src/db/schema.ts:195]: @Tadjaur I think it's a good idea to have a default value, as it's not always that we are creating this item by filling the form, especially with our latest UI changes, so I'm ok with this

File: server/src/db/schema.ts (hunk 1 above)
Comment (taronaleksanian) [reply server/src/db/schema.ts:332]: to fill existing items we need to write a script that will handle it, which will update corresponding items, so I'm ok with this @Tadjaur

File: server/src/db/schema.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikib0) [reply server/src/db/schema.ts:195]: You're right, we can't omit quantity. There is actually a [zod check](https://github.com/andrew-bierman/PackRat/blob/3c4f418b71e23a48e176f85aff15c1883e80954b/packages/validations/src/validations/itemRoutesValidator.ts#L18) to make sure it's present. I agree with you, the default will be removed and the field will be made nullable in the interim.

[the remark this answers is not in the document]
Comment (mikib0) [reply server/src/db/schema.ts:195]: > @Tadjaur I think it's a good idea to have a default value, as it's not always that we are creating this item by filling the form, especially with our latest UI changes, so I'm ok with this

Oh yeah good point. That's fine then.

**Comment by taronaleksanian** [comment]
> Global Comment:
> 
> For this kind of task, I recommend creating PR chains that represent the stages of the migration.
> 
> 1. First Pr: Add a nullable quantity field in the item_packs (server)
> 2. 2nd Pr: Import existing quantity value from item to item_packs (server)
>    ⚠️ ⚠️ ⚠️  You must always think about how to process existing data before deleting any row in a database. And if the data loss is intentional, you should always add an acknowledgement comment.
> 3. 3th Pr:
>    
>    * remove the quantity from item and make the quantity on item_packs notNull (server)
>    * apply changes on the app
> 
> Having small groups of PRs linked together will make the review simpler and easier. @andrew-bierman @taronaleksanian Please add a note if I'm wrong somewhere.

@Tadjaur  Yeah I agree, we must handle it in some phrase,  so as we have preview db, can't we just do it before release (merging development to main). So the goal of making preview db and separating preview db migrations is that, so we care about this issue only when we have ready to release version on the development.

So I'm merging this as it's only affected only preview db, we'll be more careful once we start releasing current development version

**Pull request merged**